packages/synapse-react-client/src/synapse-queries/dataaccess/useAccessApprovals.ts

```diff
@@ -1,34 +1,48 @@
-import { useInfiniteQuery, UseInfiniteQueryOptions } from 'react-query'
+import {
+  InfiniteData,
+  QueryKey,
+  useInfiniteQuery,
+  UseInfiniteQueryOptions,
+} from '@tanstack/react-query'
 import SynapseClient from '../../synapse-client'
-import { SynapseClientError } from '../../utils/SynapseClientError'
-import { useSynapseContext } from '../../utils/context/SynapseContext'
+import { SynapseClientError, useSynapseContext } from '../../utils'
 import {
   AccessApprovalSearchRequest,
   AccessApprovalSearchResponse,
 } from '@sage-bionetworks/synapse-types'
 
-export function useSearchAccessApprovalsInfinite(
+export function useSearchAccessApprovalsInfinite<
+  TData = InfiniteData<AccessApprovalSearchResponse>,
+>(
   params: AccessApprovalSearchRequest,
   options?: UseInfiniteQueryOptions<
     AccessApprovalSearchResponse,
-    SynapseClientError
+    SynapseClientError,
+    TData,
+    AccessApprovalSearchResponse,
+    QueryKey,
+    AccessApprovalSearchResponse['nextPageToken']
   >,
 ) {
   const { accessToken, keyFactory } = useSynapseContext()
-  return useInfiniteQuery<AccessApprovalSearchResponse, SynapseClientError>(
-    keyFactory.searchAccessApprovalsQueryKey(params),
-    async context => {
-      return await SynapseClient.searchAccessApprovals(
+  return useInfiniteQuery<
+    AccessApprovalSearchResponse,
+    SynapseClientError,
+    TData,
+    QueryKey,
+    AccessApprovalSearchResponse['nextPageToken']
+  >({
+    ...options,
+    queryKey: keyFactory.searchAccessApprovalsQueryKey(params),
+    queryFn: context =>
+      SynapseClient.searchAccessApprovals(
         {
           ...params,
           nextPageToken: context.pageParam,
         },
         accessToken,
-      )
-    },
-    {
-      ...options,
-      getNextPageParam: page => page.nextPageToken,
-    },
-  )
+      ),
+    initialPageParam: undefined,
```

Comment:
`initialPageParam` is now required for `useInfiniteQuery`. I think for all of our cases it's `undefined`